src/audio/mic_privacy_manager/mic_privacy_manager.c

```diff
@@ -0,0 +1,234 @@
+{
+	if (enable_irq)
+		mic_privacy_api->enable_fw_managed_irq(true, handle_fw_managed_interrupt);
+	else
+		mic_privacy_api->enable_fw_managed_irq(false, NULL);
+}
+
+void handle_dmic_interrupt(void *const self, int a, int b)
+{
+	LOG_DBG("mic_privacy handle_dmic_interrupt");
+
+	if (mic_privacy_api->get_dmic_irq_status()) {
+		uint32_t mic_disable_status = mic_privacy_api->get_dmic_mic_disable_status();
+		struct mic_privacy_settings settings = fill_mic_priv_settings(mic_disable_status);
+
+		propagate_privacy_settings(&settings);
+		mic_privacy_api->clear_dmic_irq_status();
+	}
+}
+
+void enable_dmic_irq(bool enable_irq)
+{
+	if (enable_irq)
+		mic_privacy_api->enable_dmic_irq(true, handle_dmic_interrupt);
+	else
+		mic_privacy_api->enable_dmic_irq(false, NULL);
+}
+
+void handle_fw_managed_interrupt(void * const dev)
+{
+	LOG_DBG("handle_fw_managed_interrupt");
+
+	uint32_t mic_disable_status = mic_privacy_api->get_fw_managed_mic_disable_status();
+
+	struct mic_privacy_settings mic_privacy_settings =
+					fill_mic_priv_settings(mic_disable_status);
+
+	propagate_privacy_settings(&mic_privacy_settings);
+
+	if (mic_disable_status)
+		mic_privacy_api->set_fw_mic_disable_status(true);
+	else
+		mic_privacy_api->set_fw_mic_disable_status(false);
+
+	mic_privacy_api->clear_fw_managed_irq();
+}
+
+void propagate_privacy_settings(struct mic_privacy_settings *mic_privacy_settings)
+{
+	notifier_event(mic_priv_dev, NOTIFIER_ID_MIC_PRIVACY_STATE_CHANGE,
+		       NOTIFIER_TARGET_CORE_ALL_MASK, mic_privacy_settings,
+		       sizeof(struct mic_privacy_settings));
+}
+
+uint32_t get_dma_zeroing_wait_time(void)
+{
+	return (uint32_t)mic_privacy_api->get_dma_data_zeroing_wait_time();
+}
+
+uint32_t get_privacy_mask(void)
+{
+	if (mic_privacy_policy == MIC_PRIVACY_HW_MANAGED)
+		return mic_privacy_api->get_dma_data_zeroing_link_select();
+
+	/* hardcoded for FW_MANAGED */
+	return 0xFFFFFFFF;
+}
+
+struct mic_privacy_settings fill_mic_priv_settings(uint32_t mic_disable_status)
+{
+	struct mic_privacy_settings mic_privacy_settings = {};
+
+	mic_privacy_settings.mic_privacy_mode = mic_privacy_policy;
+	mic_privacy_settings.mic_privacy_state = mic_disable_status;
+	mic_privacy_settings.privacy_mask_bits.value = get_privacy_mask();
+	mic_privacy_settings.max_ramp_time = get_dma_zeroing_wait_time();
+
+	LOG_DBG("mic_privacy_mode = %d, mic_disable_status = %d, \
+		privacy_mask = %x, max_ramp_time_in_ms = %d",
+		mic_privacy_settings.mic_privacy_mode,
+		mic_privacy_settings.mic_privacy_state,
+		mic_privacy_settings.privacy_mask_bits.value,
+		mic_privacy_settings.max_ramp_time);
+
+	return mic_privacy_settings;
+}
+
+void set_gtw_mic_state(struct mic_privacy_data *mic_priv_data,  uint32_t mic_disable_status)
+{
+	if (mic_privacy_policy == MIC_PRIVACY_HW_MANAGED) {
+		if (mic_disable_status != 0) {
+			mic_priv_data->mic_privacy_state = MIC_PRIV_MUTED;
+		} else {
+			mic_priv_data->mic_privacy_state = MIC_PRIV_UNMUTED;
+		}
```

Comment:
I'm not sure what is the coding style in SOF now-days, but this patch uses two style when there is a single instruction in the if/else
Zephyr demands curly brackets always, like this. We used to be following the kernel's with no need for them.